src/modules/quick_input_parser_module.f90

```diff
@@ -34,29 +64,34 @@ subroutine read_float_keyword(line, keyword, val)
             character(len=*), intent(in) :: keyword
             double precision :: val
```

Comment:
You should add `intent(out)` here. If assignment is optional, and `val` could have a previously defined value (e.g. max number of optimization cycles set before), then you should use `intent(inout)` here.

Reply:
> This looks like it should work. Have you tested this? If it compiles and works as expected, then I think we can merge.

Yes I tested on my side with both correct and erroneous inputs. Worked as expected.